ui/src/api/api.ts

```diff
@@ -72,6 +88,175 @@ export async function getEntities(arxivId: string, getAllEntities?: boolean) {
   return ((data as any).data || []) as Entity[];
 }
 
+// This translates from the entities-deduped API's response to the legacy
+// super-duplicated API response, as a compat shim.
+// TODO: phase this out in favor of using a more efficient data model in the reader app.
+function undedupeResponse(response: DedupedEntityResponse): EntityGetResponse {
+  // TODO: Improve error handling; this just pulls the eject handle if the response is an API error
+  if (!!(response as any).error) {
+    throw "API error: " + (response as any).error;
+  }
+  const entities: Entity[] = response.entities.map((deduped: DedupedEntity) => {
+    if (isCitation(deduped)) {
+      const citation: Citation = {
+        id: deduped.id,
+        type: 'citation',
+        attributes: {
+          bounding_boxes: deduped.attributes.bounding_boxes,
+          paper_id: deduped.attributes.paper_id,
+          source: 'tex-pipeline', // hardcoded since most everything comes out of the tex pipeline
+          tags: [],
+        },
+        relationships: {},
+      };
+      return citation;
+    } else if (isEquation(deduped)) {
+      const equation: Equation = {
+        id: deduped.id,
+        type: 'equation',
+        attributes: {
+          bounding_boxes: deduped.attributes.bounding_boxes,
+          source: 'tex-pipeline', // hardcoded since most everything comes out of the tex pipeline
+          tags: [],
+          tex: deduped.attributes.tex,
+        },
+        relationships: {},
+      };
+      return equation;
+    } else if (isSentence(deduped)) {
+      const sentence: Sentence = {
+        id: deduped.id,
+        type: 'sentence',
+        attributes: {
+          bounding_boxes: deduped.attributes.bounding_boxes,
+          source: 'tex-pipeline', // hardcoded since most everything comes out of the tex pipeline
+          tags: [],
+          tex: deduped.attributes.tex,
+          tex_start: deduped.attributes.tex_start,
+          tex_end: deduped.attributes.tex_end,
+          text: deduped.attributes.text,
+        },
+        relationships: {},
+      };
+      return sentence;
+    } else if (isSymbol(deduped)) {
+      const mathml = deduped.attributes.disambiguated_id;
```

Comment:
We should probably call this `const disambiguatedId` because I think the connection to `mathml` is likely to change if/when Andrew does real symbol disambiguation in the future temporary.